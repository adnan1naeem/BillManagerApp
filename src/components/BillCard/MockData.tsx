const MockData = [
  {
    month: "Jan",
    budget: 2000,
    name:"Bills"
  },
  {
    month: "Jan",
    budget: 2000,
    name:"Bills"
  },{
    month: "Jan",
    budget: 2000,
    name:"Bills"
  }
  
]



