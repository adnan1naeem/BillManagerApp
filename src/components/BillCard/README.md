#  BillCard Component

## Compatibility

| 🌏 Web                   | 🖥 Electron              | 📱 React Native |
| :----------------------: | :----------------------: | :-------------: |
| :heavy_multiplication_x: | :heavy_multiplication_x: | ::              |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native                         |
| :----: | :---------: | :-------------------------------------: |
| TBD    | TBD         | ![BillCard](./Screenshots/BillCard.png) |

## Props

| Name   | Type   | Default | Description                           |
| :----- | :----- | :------ | :------------------------------------ |
| name   | string |         | It is used for adding billName        |
| month  | string |         | It is used for adding month of bill   |
| budget | number |         | It is used for adding budget of bills |
| style  | object |         | It is used for adding stylesin AppBar |


## How to use

```react
import React from 'react';
<BillCard
      name={"ADNAN "}
      month={"Jan"}
      budget={2000}
    />
```

## Tests

