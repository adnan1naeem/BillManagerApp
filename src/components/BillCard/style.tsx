export const style = {
	card: {

	},
	link: {
		textDecoration: 'none'
	},
	text: {
		fontSize: '14px',
		overflow: 'hidden',
		textOverflow: 'ellipsis',
		whiteSpace: 'nowrap',
		display: 'block',
		color: '#373a3c'
	},
	container: {
		width: 150,
		paddingTop: 10,
		paddingBottom: 10
	},
	body: {

		backgroundColor: 'white'
	},

	img: {
		height: 180,

		opacity: '0.2'
	},
	cardBody: {
		display: 'flex',
		flexDirection: 'column',
	},
	topView: {
		height: 180,
		backgroundColor: '#0093FF',
	},
	descriptionView: {
		display: 'flex',
		flexDirection: 'column',
		padding: 7
	},
	consumedLabel: {
		color: 'white',
		fontSize: 12,
	},
	budgetContanier: {
		display: 'flex',
		flexDirection: 'row'
	},
	budget: {
		color: 'white',
		fontSize: 20,
		flex: 1
	},

	lowerView: {
		//s	backgroundColor:"#DCDCDC",
		display: 'flex',
		padding: 7,
		flexDirection: 'column',
	},
	name: {
		color: '#0093FF',
		fontSize: 14,
		overflow: 'hidden',
	},
	monthContainer: {
		display: 'flex',
		flexDirection: 'row',

	},
	month: {
		color: '#0093FF',
		fontSize: 12,
		flex: 1

	},
};