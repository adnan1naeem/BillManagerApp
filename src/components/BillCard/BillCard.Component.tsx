import React from 'react'
import View from "../View/View.Component";
import Text from "../Text/Text.Component";
import { style } from "./style"

export interface BillCardProperties {
	name: string,
	month: string,
	budget: number,
	Style?: any,
}

const BillCard =(props: BillCardProperties) => {

	const { name, month, budget } = props;

	return (
		<View style={style.container}>
			<View style={style.topView}>
				<View style={style.descriptionView}>
					<Text style={style.consumedLabel}>
						Total Consumed
					</Text>
					<View style={style.budgetContanier}>
						<Text style={style.budget}>
							Rs. {budget}
						</Text>
					</View>
				</View>
			</View>
			<View style={style.body}>
				<View style={style.lowerView}>
					<Text style={style.name}>
						{name}
					</Text>
					<View style={style.monthContainer}>
						<Text style={style.month}>
							{month}
						</Text>
					</View>
				</View>
			</View>
		</View>
	);
};

export default BillCard;