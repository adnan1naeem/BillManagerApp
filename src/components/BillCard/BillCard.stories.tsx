import React from 'react';
import { storiesOf } from '@storybook/react-native';
import BillCard from './BillCard.component';
import { withKnobs} from '@storybook/addon-knobs/react';
import {View} from 'react-native';


storiesOf('Card', module)
  .addDecorator(getStory => <View>{getStory()}</View>)
  .addDecorator(withKnobs )
  .add('Card Story', () => (
  <BillCard  
  name="ADNAN"
	month="June"
	budget={22222}
  />
  ))

  