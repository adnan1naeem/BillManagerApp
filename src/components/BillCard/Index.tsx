import BillCard from  "./BillCard.component";
export default BillCard; 