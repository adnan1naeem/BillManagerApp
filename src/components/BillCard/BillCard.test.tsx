import * as React from 'react'
import { shallow, configure, mount } from 'enzyme';
import BillCard from "./BillCard.component";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';


configure({ adapter: new Adapter() })

describe('', () => {


  test('renders correctly', () => {
    shallow(<BillCard budget={234} month="January" name="Okay Budget" />);
  });


  test('renders correctly', () => {
    const tree = renderer.create(<BillCard budget={234} month="January" name="Okay Budget" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

