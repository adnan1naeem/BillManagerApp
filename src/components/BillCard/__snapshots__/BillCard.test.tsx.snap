// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": 150,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#0093FF",
        "height": 180,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "padding": 7,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 12,
          }
        }
      >
        Total Consumed
      </Text>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 20,
            }
          }
        >
          Rs.
           
          234
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "padding": 7,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#0093FF",
            "fontSize": 14,
            "overflow": "hidden",
          }
        }
      >
        Okay Budget
      </Text>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#0093FF",
              "flex": 1,
              "fontSize": 12,
            }
          }
        >
          January
        </Text>
      </View>
    </View>
  </View>
</View>
`;
